graph.py

```diff
@@ -0,0 +1,77 @@
+
+class SimpleGraph(object):
+    '''This is a simple graph program that will allow us
+                to impliment a graph data structure'''
+    def __init__(self, dict_graph={}):
+        self.dict_graph = dict_graph
+
+    def node(self):
+        '''return a list of all nodes in the graph'''
+        return self.dict_graph.keys()
+
+    def edges(self):
+        '''return a list of all edges in the graph'''
+        edges = ''
+        for key, value in self.dict_graph.items():
+            edges = '{}->{}  {}'.format(key, value, edges)
+        return edges
+
+    def add_node(self, n):
+        '''adds a new node 'n' to the graph'''
+        if n not in self.dict_graph:
+            self.dict_graph[n] = []
+
+    def add_edge(self, n1, n2):
+        '''adds a new edge to the graph connecting 'n1' and 'n2',
+        if either n1 or n2 are not already present in the graph,
+        they should be added.'''
+        if n1 in self.dict_graph:
+            self.dict_graph[n1].append(n2)
+            if n2 not in self.dict_graph:
+                self.add_node(n2)
+        else:
+            self.dict_graph[n1] = [n2]
+            if n2 not in self.dict_graph:
+                self.add_node(n2)
```

Comment:
You have the same issues of pythonicity as we discussed in class today.  This could be significantly cleaner than it is.